ngraph/test/util/test_environment.cpp

```diff
@@ -0,0 +1,136 @@
+            return *p_instance;
+        }
+
+        void Summary::updateOPsStats(ngraph::NodeTypeInfo op, PassRate::Statuses status)
+        {
+            // TODO: Do we need to count skips?
+            auto it = opsStats.find(op);
+            if (it != opsStats.end())
+            {
+                auto& passrate = it->second;
+                switch (status)
+                {
+                case PassRate::PASSED: passrate.passed += 1; break;
+                case PassRate::FAILED: passrate.failed += 1; break;
+                case PassRate::SKIPPED: passrate.skipped += 1; break;
+                }
+            }
+            else
+            {
+                switch (status)
+                {
+                case PassRate::PASSED: opsStats[op] = PassRate(1, 0, 0); break;
+                case PassRate::FAILED: opsStats[op] = PassRate(0, 1, 0); break;
+                case PassRate::SKIPPED: opsStats[op] = PassRate(0, 0, 1); break;
+                }
+            }
+        }
+
+        void TestEnvironment::TearDown()
+        {
+            std::vector<ngraph::OpSet> opsets;
+            opsets.push_back(ngraph::get_opset1());
+            opsets.push_back(ngraph::get_opset2());
+            opsets.push_back(ngraph::get_opset3());
+            opsets.push_back(ngraph::get_opset4());
+            opsets.push_back(ngraph::get_opset5());
+            std::set<ngraph::NodeTypeInfo> opsInfo;
+            for (const auto& opset : opsets)
+            {
+                const auto& type_info_set = opset.get_type_info_set();
+                opsInfo.insert(type_info_set.begin(), type_info_set.end());
+            }
+
+            auto& s = Summary::getInstance();
+            auto stats = s.getOPsStats();
+
+            pugi::xml_document doc;
+
+            std::ifstream file;
+            file.open(reportFileName);
+
+            time_t rawtime;
+            struct tm* timeinfo;
+            char timeNow[80];
+
+            time(&rawtime);
+            // cpplint require to use localtime_r instead which is not available in C++14
+            timeinfo = localtime(&rawtime); // NOLINT
+
+            strftime(timeNow, sizeof(timeNow), "%d-%m-%Y %H:%M:%S", timeinfo);
+
+            pugi::xml_node root;
+            if (file)
+            {
+                doc.load_file(reportFileName.c_str());
+                root = doc.child("report");
+                // Ugly but shorter than to write predicate for find_atrribute() to update existing
+                // one
+                root.remove_attribute("timestamp");
+                root.append_attribute("timestamp").set_value(timeNow);
+
+                root.remove_child("ops_list");
+                root.child("results").remove_child(s.deviceName.c_str());
+            }
+            else
+            {
+                root = doc.append_child("report");
+                root.append_attribute("timestamp").set_value(timeNow);
+                root.append_child("results");
+            }
```

Comment:
If we want to overwrite a report maybe we should remove the old one and create a new one instead of modifying the existing one?